### DISCUSSION TITLE
Bringing vars back to command/host env

### DISCUSSION THREAD

**Discussion by paul-hammant** [madflojo/efs2#11] [Ideas]
## Describe the idea

The commanding machine might like to retrieve one or more vars from a remote invoked Efs2file script. Like so:

```
CAPTURE $vm_pub_key
RUN cat /root/.ssh/id_rsa.pub

PUT something_else somewhere_else
RUN something
```

The above would be for VMs created with initial name/password entry .. and an Efs2file script would upgrade that to PKI in the target machine.  

## Additional context

Sure, the entire log of an Efs2file invocation could be captured and then grep/sed'd on the host* to extract the same info, but that could result in many smaller Efs2file rather than a reduced number that could casually capture multiple vars as they go.

\* what's the canonical name for the controlling machine: host, hub, controller ??

**Comment by madflojo** [comment]
I have to think about this. I like the idea of doing variables within the Efs2file but need to think through a way to implement it that still keeps things simple.

**Comment by paul-hammant** [comment]
You know, it'd be OK to capture vars as the script progresses, then return them from guest/target to host/source in one go. Specifically, the last operation the executor does is to return vars as a set.

**Comment by madflojo** [comment]
Yeah, right now the file is fully parsed then each command executes on its own. I would need to change how it parses and specifically split it for each host. Again I’m with the idea just want to think about how to do it before I implement.

**Comment by paul-hammant** [comment]
I think that's OK too. Traction before perfection.
